catalog/resource_volumes.go

```diff
@@ -0,0 +1,104 @@
+package catalog
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// This describes an enum
+type VolumeType string
+
+type VolumesAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+type CreateVolumeRequestContent struct {
+	Name            string     `json:"name"`
+	CatalogName     string     `json:"catalog_name"`
+	VolumeType      VolumeType `json:"volume_type"`
+	SchemaName      string     `json:"schema_name"`
+	StorageLocation string     `json:"storage_location,omitempty"`
+	Comment         string     `json:"comment,omitempty"`
+}
+
+type UpdateVolumeRequestContent struct {
+	Comment string `json:"comment,omitempty"`
+	Name    string `json:"name,omitempty"`
+	Owner   string `json:"owner,omitempty"`
+}
+
+type ListVolumesResponseContent struct {
+	Volumes *VolumeInfo `json:"volumes,omitempty"`
+}
+
+func NewVolumesAPI(ctx context.Context, m any) VolumesAPI {
+	return VolumesAPI{m.(*common.DatabricksClient), context.WithValue(ctx, common.Api, common.API_2_1)}
+}
+
+// TODO: check all for tf:"computed", for eg: it is probably required in created at/by, updated at/by
+type VolumeInfo struct {
+	Owner           string     `json:"owner,omitempty"`
+	StorageLocation string     `json:"storage_location,omitempty"`
+	VolumeId        string     `json:"volume_id,omitempty"`
+	CatalogName     string     `json:"catalog_name,omitempty"`
+	SchemaName      string     `json:"schema_name,omitempty"`
+	VolumeType      VolumeType `json:"volume_type,omitempty"`
+	MetaStoreId     string     `json:"metastore_id,omitempty"`
+	Name            string     `json:"name,omitempty"`
+	FullName        string     `json:"full_name,omitempty"`
+	CreatedBy       string     `json:"created_by,omitempty"`
+	CreatedAt       float64    `json:"created_at,omitempty"`
+	UpdatedBy       string     `json:"updated_by,omitempty"`
+	UpdatedAt       float64    `json:"updated_at,omitempty"`
+	Comment         string     `json:"comment,omitempty"`
+}
+
+func (a VolumesAPI) create(v *VolumeInfo) error {
+	return a.client.Post(a.context, "/unity-catalog/volumes", v, &v)
+}
+
+func (a VolumesAPI) get(name string) (v VolumeInfo, err error) {
+	err = a.client.Get(a.context, "/unity-catalog/volumes/"+url.PathEscape(name), nil, &v)
+	return
+}
+
+func (a VolumesAPI) delete(name string) error {
+	return a.client.Delete(a.context, "/unity-catalog/volumes/"+url.PathEscape(name), nil)
+}
+
+func ResourceVolumes() *schema.Resource {
+	s := common.StructToSchema(VolumeInfo{},
```

Comment:
use `catalog.UpdateVolumeRequestContent` as the schema for the request: https://pkg.go.dev/github.com/databricks/databricks-sdk-go@v0.8.1/service/catalog#UpdateVolumeRequestContent

Reply:
I don't think it is possible to move a volume to a new catalog & schema, which is why these parameters are not part of the update request